pkg/apis/v1alpha2/aksnodeclass_hash_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2_test
+
+import (
+	"time"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/imdario/mergo"
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/karpenter/pkg/test"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Hash", func() {
+	const staticHash = "3322684356700793451"
+	var nodeClass *v1alpha2.AKSNodeClass
+	BeforeEach(func() {
+		nodeClass = &v1alpha2.AKSNodeClass{
+			ObjectMeta: test.ObjectMeta(metav1.ObjectMeta{}),
+			Spec: v1alpha2.AKSNodeClassSpec{
+				VNETSubnetID: lo.ToPtr("subnet-id"),
+				OSDiskSizeGB: lo.ToPtr(int32(30)),
+				ImageFamily:  lo.ToPtr("Ubuntu2204"),
+				Tags: map[string]string{
+					"keyTag-1": "valueTag-1",
+					"keyTag-2": "valueTag-2",
+				},
+				Kubelet: &v1alpha2.KubeletConfiguration{
+					CPUManagerPolicy:            "static",
+					CPUCFSQuota:                 lo.ToPtr(true),
+					CPUCFSQuotaPeriod:           metav1.Duration{Duration: lo.Must(time.ParseDuration("100ms"))},
+					ImageGCHighThresholdPercent: lo.ToPtr(int32(85)),
+					ImageGCLowThresholdPercent:  lo.ToPtr(int32(80)),
+					TopologyManagerPolicy:       "none",
+					AllowedUnsafeSysctls:        []string{"net.core.somaxconn"},
+					ContainerLogMaxSize:         "10Mi",
+					ContainerLogMaxFiles:        lo.ToPtr(int32(10)),
+				},
+				MaxPods: lo.ToPtr(int32(100)),
+			},
+		}
+	})
+	DescribeTable(
+		"should match static hash on field value change",
```

Comment:
All of these hashes will have to be updated, if we add more fields we hash on, correct?

Please as a comment here noting this, so when expected failures occur in the future, we know these needs to be updated. Something like:
```
// NOTE: When the hashing algorithm updates, these tests are expected to fail and need to have their hashes updated.
```